src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -290,6 +290,73 @@ class HMM
    */
   double LogLikelihood(const arma::mat& dataSeq) const;
 
+  /**
+   * Compute the log of the scaling factor of the given emission probability
+   * at time t. To calculate the log-likelihood for the whole sequence,
+   * accumulate log scale over the entire sequence
+   *
+   * @param emissionLogProb emission probability at time t.
+   * probability up to time t-1
+   * @param prevForwardLogProb Vector in which forward probabilities for time
+   * t-1 will be saved. Passing prevForwardProb as an empty vector indicates the
+   * start of sequence or time t=0
+   * @param forwardLogProb Vector in which forward probabilities for time t
+   * will be saved.
+   * @return Log scale factor of the given sequence of emission at time t.
+   */
+  double LogScaleEmissionProb(const arma::vec& emissionLogProb,
+                       arma::vec& prevForwardLogProb,
+                       arma::vec& forwardLogProb) const;
```

Comment:
Same question as below here---is this something we should provide to the user API, or should we make it private?